File: Backend/src/main/java/pw/react/carly/car/Car.java
package pw.react.carly.car;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Entity(name = "Cars")
@Table(name = "Cars")
public class Car {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotBlank
    private String model;
    @NotBlank
    private String make;
    @NotNull
    private int seats;
    @NotNull
    private int year;
    @NotBlank
    private String licence;
    @NotBlank
    private String location;
    @NotNull
    @DecimalMin(value = "0.0", inclusive = true)
    private BigDecimal price;
    @JsonIgnore
    private boolean isActive;

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean active) {
        isActive = active;
    }

    public Car(String model, String make, int seats, int year, String licence, String location, BigDecimal price) {
        this.model = model;
        this.make = make;
        this.seats = seats;
        this.year = year;
        this.licence = licence;
        this.location = location;
        this.price = price;
        this.isActive = true;
    }

    public Car(){
        this.isActive = true;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getMake() {
        return make;
    }

    public void setMake(String make) {
        this.make = make;
    }

    public int getSeats() {
        return seats;
    }

    public void setSeats(int seats) {
        this.seats = seats;
    }



    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public String getLicence() {
        return licence;
    }

    public void setLicence(String licence) {
        this.licence = licence;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

}

File: Web/src/Models/Car.ts



export class Car {
    id: string;
    carMake: string;
    carModel: string;
    licenseNumber: string;
    seats: number;
    year: number;
    location: string;

    constructor(data: any = {}) {
        if (!data)
            return;
        this.id = data.id ?? null;
        this.carMake = data.carMake ?? null;
        this.carModel = data.carModel ?? null;
        this.licenseNumber = data.licenseNumber ?? null;//TYPO
        this.seats = data.seats ?? null;
        this.year = data.year ?? null;
        this.location = data.location ?? null;
    }
    static parseData(data: any[]): Car[] {
        var cars: Car[] = [];
        if (!data)
            return [];
        try {
            data.forEach(x => {
                cars.push(new Car(
                    {
                        id: x.id,
                        carModel: x.model,
                        carMake: x.make,
                        seats: x.seats,
                        year: x.year,
                        licenseNumber: x.licence,
                        location: x.location,
                    }
                )
                )
            });
        }
        catch (e) {

            console.log('parse car exception')
            console.log(e)
        }
        return cars;
    }
}